agent/replacecall/src/main/java/randoop/instrument/ReplacementFileReader.java

```diff
@@ -0,0 +1,556 @@
+              replacementPackage + filename,
+              subdirectory);
+        }
+      }
+    }
+  }
+
+  /**
+   * For each class of the replacement package in the given jar file, calls {@link
+   * #addReplacementsForClass(ConcurrentHashMap, String, String)} to add method replacements from
+   * the original package to the replacement package.
+   *
+   * @param replacementMap the method replacement map to which new replacements are added
+   * @param originalPackage the original package name
+   * @param replacementPackage the replacement package name
+   * @param jarFile the jar file to search
+   * @see #addReplacementsForPackage(ConcurrentHashMap, String, String)
+   * @see #addReplacementsForPackage(ConcurrentHashMap, String, String, ClassLoader)
+   */
+  private static void addReplacementsFromAllClassesOfPackage(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      String originalPackage,
+      String replacementPackage,
+      JarFile jarFile)
+      throws ReplacementException, ClassNotFoundException {
+    String replacementPath = replacementPackage.replace('.', '/') + "/";
+    Enumeration<JarEntry> entries = jarFile.entries();
+    while (entries.hasMoreElements()) {
+      JarEntry entry = entries.nextElement();
+      String filename = entry.getName();
+      if (filename.endsWith(".class") && filename.startsWith(replacementPath)) {
+        final String classname =
+            filename.substring(replacementPackage.length() + 1, filename.lastIndexOf(".class"));
+        final String originalClassname = originalPackage + "." + classname;
+        final String replacementClassname = replacementPackage + "." + classname;
+        addReplacementsForClass(replacementMap, originalClassname, replacementClassname);
+      }
+    }
+  }
+
+  /**
+   * Adds replacements for methods in the original class, if it exists, to methods found in the
+   * replacement class.
+   *
+   * @param replacementMap the map to which replacements are added
+   * @param originalClassname the name of the original class
+   * @param replacementClassname the name of the replacement class
+   * @throws ClassNotFoundException if either the original or replacement class cannot be loaded
+   */
+  private static void addReplacementsForClass(
+      ConcurrentHashMap<MethodSignature, MethodSignature> replacementMap,
+      String originalClassname,
+      String replacementClassname)
+      throws ClassNotFoundException, ReplacementException {
+
+    // Check that original class exists
+    Class<?> originalClass;
+    originalClass = Class.forName(originalClassname);
+
+    if (originalClass != null) {
+      Class<?> replacementClass = Class.forName(replacementClassname);
+      addReplacementsForClass(replacementMap, originalClassname, replacementClass);
+    }
+  }
+
+  /** Exception to represent a malformed replacement file. */
+  private static class ReplacementException extends Throwable {
```

Comment:
Why do both `ReplacementException` and `ReplacementFileException` exist?  The Javadoc comment here suggests that they are both about the file and should be unified, so that only `ReplacementFileException` (I guess) should exist.